{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+--------------------+--------------+--------------------+-----------------------+--------------------+\n",
      "|                path|         label|            features|standard_scale_features|        pca_features|\n",
      "+--------------------+--------------+--------------------+-----------------------+--------------------+\n",
      "|s3://ngor-yoff/Te...|    Watermelon|[0.01487588509917...|   [-0.7179758847444...|[-6.1051797642081...|\n",
      "|s3://ngor-yoff/Te...|    Watermelon|[0.07104229182004...|   [-0.6083942088385...|[-5.1358059500146...|\n",
      "|s3://ngor-yoff/Te...|Pineapple Mini|[0.0,4.1723289489...|   [-0.7469990057685...|[-10.883470681318...|\n",
      "|s3://ngor-yoff/Te...|Pineapple Mini|[0.0,4.6597709655...|   [-0.7469990057685...|[-5.1053482585690...|\n",
      "|s3://ngor-yoff/Te...|    Watermelon|[0.00825151987373...|   [-0.7309001411744...|[-3.4430835379807...|\n",
      "|s3://ngor-yoff/Te...|    Watermelon|[0.0,0.2746154069...|   [-0.7469990057685...|[-0.5512291179439...|\n",
      "|s3://ngor-yoff/Te...|     Raspberry|[0.03869540989398...|   [-0.6715035608289...|[-1.6638116032262...|\n",
      "|s3://ngor-yoff/Te...|   Cauliflower|[0.0,0.5575997233...|   [-0.7469990057685...|[-3.7828199000510...|\n",
      "|s3://ngor-yoff/Te...|   Cauliflower|[0.0,0.8316052556...|   [-0.7469990057685...|[-3.1840693594234...|\n",
      "|s3://ngor-yoff/Te...|     Pineapple|[0.0,3.5644376277...|   [-0.7469990057685...|[-9.5307185021043...|\n",
      "|s3://ngor-yoff/Te...|     Pineapple|[0.00830239988863...|   [-0.7308008733446...|[-9.6648524992022...|\n",
      "|s3://ngor-yoff/Te...|     Pineapple|[0.0,4.0233659744...|   [-0.7469990057685...|[-8.4923513999398...|\n",
      "|s3://ngor-yoff/Te...|   Cauliflower|[0.0,0.3262555003...|   [-0.7469990057685...|[-6.6387534473007...|\n",
      "|s3://ngor-yoff/Te...|Pineapple Mini|[0.0,4.8278598785...|   [-0.7469990057685...|[-8.5426058640954...|\n",
      "|s3://ngor-yoff/Te...|   Onion White|[0.0,0.0442409664...|   [-0.7469990057685...|[-8.5924177912406...|\n",
      "|s3://ngor-yoff/Te...|Pineapple Mini|[0.0,4.7213129997...|   [-0.7469990057685...|[-8.9211728458688...|\n",
      "|s3://ngor-yoff/Te...|     Raspberry|[0.47354778647422...|   [0.17690129501549...|[2.92074957361591...|\n",
      "|s3://ngor-yoff/Te...|   Onion White|[0.06173849105834...|   [-0.6265460924871...|[-2.7786745741520...|\n",
      "|s3://ngor-yoff/Te...|        Lychee|[1.10436165332794...|   [1.40763056452420...|[-8.6116628851722...|\n",
      "|s3://ngor-yoff/Te...|        Lychee|[0.59805363416671...|   [0.41981445693505...|[-6.1353041583971...|\n",
      "+--------------------+--------------+--------------------+-----------------------+--------------------+\n",
      "only showing top 20 rows\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from pyspark.sql import SparkSession\n",
    "\n",
    "# Initialisation de la session Spark\n",
    "spark = SparkSession.builder \\\n",
    "    .appName(\"Data Processing with Spark\") \\\n",
    "    .getOrCreate()\n",
    "\n",
    "# Lecture des fichiers Parquet depuis le dossier spécifié\n",
    "input_path = '/home/ibugueye/livrables/ngor_yoff_results_last/*.parquet'\n",
    "df = spark.read.parquet(input_path)\n",
    "\n",
    "# Exemple de traitement : Afficher les premières lignes du DataFrame\n",
    "df.show()\n",
    "\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">/tmp/ipykernel_34897/1423860683.py:</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">2</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\"> DeprecationWarning</span><span style=\"color: #808000; text-decoration-color: #808000\">: The `airflow.operators.bash_operator.BashOperator` class is deprecated. Please use `</span><span style=\"color: #808000; text-decoration-color: #808000\">'airflow.operators.bash.BashOperator'</span><span style=\"color: #808000; text-decoration-color: #808000\">`.</span>\n",
       "</pre>\n"
      ],
      "text/plain": [
       "\u001b[1;33m/tmp/ipykernel_34897/\u001b[0m\u001b[1;33m1423860683.py\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m2\u001b[0m\u001b[1;33m DeprecationWarning\u001b[0m\u001b[33m: The `airflow.operators.bash_operator.BashOperator` class is deprecated. Please use `\u001b[0m\u001b[33m'airflow.operators.bash.BashOperator'\u001b[0m\u001b[33m`.\u001b[0m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">/tmp/ipykernel_34897/1423860683.py:</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">13</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\"> RemovedInAirflow3Warning</span><span style=\"color: #808000; text-decoration-color: #808000\">: Param `schedule_interval` is deprecated and will be removed in a future release. Please use `schedule` instead.</span>\n",
       "</pre>\n"
      ],
      "text/plain": [
       "\u001b[1;33m/tmp/ipykernel_34897/\u001b[0m\u001b[1;33m1423860683.py\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m13\u001b[0m\u001b[1;33m RemovedInAirflow3Warning\u001b[0m\u001b[33m: Param `schedule_interval` is deprecated and will be removed in a future release. Please use `schedule` instead.\u001b[0m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from airflow import DAG\n",
    "from airflow.operators.bash_operator import BashOperator\n",
    "from datetime import datetime\n",
    "\n",
    "# Définir les arguments par défaut\n",
    "default_args = {\n",
    "    'owner': 'airflow',\n",
    "    'start_date': datetime(2024, 5, 30),\n",
    "    'retries': 1,\n",
    "}\n",
    "\n",
    "# Initialisation du DAG\n",
    "dag = DAG('spark_processing_dag', default_args=default_args, schedule_interval='@daily')\n",
    "\n",
    "\n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">/tmp/ipykernel_34897/2369173825.py:</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">2</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\"> DeprecationWarning</span><span style=\"color: #808000; text-decoration-color: #808000\">: The `airflow.operators.bash_operator.BashOperator` class is deprecated. Please use `</span><span style=\"color: #808000; text-decoration-color: #808000\">'airflow.operators.bash.BashOperator'</span><span style=\"color: #808000; text-decoration-color: #808000\">`.</span>\n",
       "</pre>\n"
      ],
      "text/plain": [
       "\u001b[1;33m/tmp/ipykernel_34897/\u001b[0m\u001b[1;33m2369173825.py\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m2\u001b[0m\u001b[1;33m DeprecationWarning\u001b[0m\u001b[33m: The `airflow.operators.bash_operator.BashOperator` class is deprecated. Please use `\u001b[0m\u001b[33m'airflow.operators.bash.BashOperator'\u001b[0m\u001b[33m`.\u001b[0m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\"><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">/tmp/ipykernel_34897/2369173825.py:</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\">13</span><span style=\"color: #808000; text-decoration-color: #808000; font-weight: bold\"> RemovedInAirflow3Warning</span><span style=\"color: #808000; text-decoration-color: #808000\">: Param `schedule_interval` is deprecated and will be removed in a future release. Please use `schedule` instead.</span>\n",
       "</pre>\n"
      ],
      "text/plain": [
       "\u001b[1;33m/tmp/ipykernel_34897/\u001b[0m\u001b[1;33m2369173825.py\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m13\u001b[0m\u001b[1;33m RemovedInAirflow3Warning\u001b[0m\u001b[33m: Param `schedule_interval` is deprecated and will be removed in a future release. Please use `schedule` instead.\u001b[0m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from airflow import DAG\n",
    "from airflow.operators.bash_operator import BashOperator\n",
    "from datetime import datetime\n",
    "\n",
    "# Définir les arguments par défaut\n",
    "default_args = {\n",
    "    'owner': 'airflow',\n",
    "    'start_date': datetime(2024, 5, 30),\n",
    "    'retries': 1,\n",
    "}\n",
    "\n",
    "# Initialisation du DAG\n",
    "dag = DAG('spark_processing_dag', default_args=default_args, schedule_interval='@daily')\n",
    "\n",
    "# Tâche pour exécuter le script Spark\n",
    "spark_task = BashOperator(\n",
    "    task_id='run_spark_script',\n",
    "    bash_command='python /home/ibugueye/livrables/spark_script.py',  # Remplacez par le chemin réel de votre script\n",
    "    dag=dag,\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Please confirm database initialize (or wait 4 seconds to skip it). Are you sure? [y/N]\n",
      "  ____________       _____________\n",
      " ____    |__( )_________  __/__  /________      __\n",
      "____  /| |_  /__  ___/_  /_ __  /_  __ \\_ | /| / /\n",
      "___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /\n",
      " _/_/  |_/_/  /_/    /_/    /_/  \\____/____/|__/\n",
      "Running the Gunicorn Server with:\n",
      "Workers: 4 sync\n",
      "Host: 0.0.0.0:8080\n",
      "Timeout: 120\n",
      "Logfiles: - -\n",
      "Access Logformat: \n",
      "=================================================================\n",
      "\u001b[1;33m/home/ibugueye/.local/lib/python3.10/site-packages/flask_limiter/\u001b[0m\u001b[1;33mextension.py\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m333\u001b[0m\u001b[1;33m UserWarning\u001b[0m\u001b[33m: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: \u001b[0m\u001b[4;33mhttps://flask-limiter.readthedocs.io#configuring-a-storage-backend\u001b[0m\u001b[33m for documentation about configuring the storage backend.\u001b[0m\n",
      "[\u001b[34m2024-05-31T18:19:19.182+0200\u001b[0m] {\u001b[34moverride.py:\u001b[0m924} \u001b[33mWARNING\u001b[0m - \u001b[33mNo user yet created, use flask fab command to do it.\u001b[0m\n",
      "[2024-05-31 18:19:21 +0200] [26165] [INFO] Starting gunicorn 22.0.0\n",
      "[2024-05-31 18:19:21 +0200] [26165] [INFO] Listening at: http://0.0.0.0:8080 (26165)\n",
      "[2024-05-31 18:19:21 +0200] [26165] [INFO] Using worker: sync\n",
      "[2024-05-31 18:19:21 +0200] [26171] [INFO] Booting worker with pid: 26171\n",
      "[2024-05-31 18:19:21 +0200] [26172] [INFO] Booting worker with pid: 26172\n",
      "[2024-05-31 18:19:21 +0200] [26173] [INFO] Booting worker with pid: 26173\n",
      "[2024-05-31 18:19:21 +0200] [26174] [INFO] Booting worker with pid: 26174\n",
      "[\u001b[34m2024-05-31T18:19:34.372+0200\u001b[0m] {\u001b[34mapp.py:\u001b[0m1744} \u001b[31mERROR\u001b[0m - \u001b[31mException on / [GET]\u001b[0m\n",
      "\u001b[31mTraceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2529, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1826, in full_dispatch_request\n",
      "    return self.finalize_request(rv)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ffcb2537-3f35-4674-8ffd-5956a07b2132', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\u001b[0m\n",
      "[\u001b[34m2024-05-31T18:19:34.401+0200\u001b[0m] {\u001b[34mapp.py:\u001b[0m1852} \u001b[31mERROR\u001b[0m - \u001b[31mRequest finalizing failed with an error while handling an error\u001b[0m\n",
      "\u001b[31mTraceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2529, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1826, in full_dispatch_request\n",
      "    return self.finalize_request(rv)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ffcb2537-3f35-4674-8ffd-5956a07b2132', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ffcb2537-3f35-4674-8ffd-5956a07b2132', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\u001b[0m\n",
      "127.0.0.1 - - [31/May/2024:18:19:34 +0200] \"GET / HTTP/1.1\" 500 1592 \"-\" \"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36\"\n",
      "[\u001b[34m2024-05-31T18:19:34.620+0200\u001b[0m] {\u001b[34mapp.py:\u001b[0m1744} \u001b[31mERROR\u001b[0m - \u001b[31mException on /favicon.ico [GET]\u001b[0m\n",
      "\u001b[31mTraceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2529, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1826, in full_dispatch_request\n",
      "    return self.finalize_request(rv)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ae400c5a-aa5b-4be1-888f-9e2e78dc88a8', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\u001b[0m\n",
      "[\u001b[34m2024-05-31T18:19:34.631+0200\u001b[0m] {\u001b[34mapp.py:\u001b[0m1852} \u001b[31mERROR\u001b[0m - \u001b[31mRequest finalizing failed with an error while handling an error\u001b[0m\n",
      "\u001b[31mTraceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2529, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1826, in full_dispatch_request\n",
      "    return self.finalize_request(rv)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ae400c5a-aa5b-4be1-888f-9e2e78dc88a8', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlite3.OperationalError: no such table: session\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 1847, in finalize_request\n",
      "    response = self.process_response(response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask/app.py\", line 2344, in process_response\n",
      "    self.session_interface.save_session(self, ctx.session, response)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/airflow/www/session.py\", line 33, in save_session\n",
      "    return super().save_session(*args, **kwargs)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/flask_session/sessions.py\", line 544, in save_session\n",
      "    session_id=store_id).first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2824, in first\n",
      "    return self.limit(1)._iter().first()\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py\", line 2916, in _iter\n",
      "    result = self.session.execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 1717, in execute\n",
      "    result = conn._execute_20(statement, params or {}, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1710, in _execute_20\n",
      "    return meth(self, args_10style, kwargs_10style, execution_options)\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py\", line 334, in _execute_on_connection\n",
      "    return connection._execute_clauseelement(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1577, in _execute_clauseelement\n",
      "    ret = self._execute_context(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1953, in _execute_context\n",
      "    self._handle_dbapi_exception(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 2134, in _handle_dbapi_exception\n",
      "    util.raise_(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 211, in raise_\n",
      "    raise exception\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py\", line 1910, in _execute_context\n",
      "    self.dialect.do_execute(\n",
      "  File \"/home/ibugueye/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py\", line 736, in do_execute\n",
      "    cursor.execute(statement, parameters)\n",
      "sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session\n",
      "[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry \n",
      "FROM session \n",
      "WHERE session.session_id = ?\n",
      " LIMIT ? OFFSET ?]\n",
      "[parameters: ('ae400c5a-aa5b-4be1-888f-9e2e78dc88a8', 1, 0)]\n",
      "(Background on this error at: https://sqlalche.me/e/14/e3q8)\u001b[0m\n",
      "127.0.0.1 - - [31/May/2024:18:19:34 +0200] \"GET /favicon.ico HTTP/1.1\" 500 1592 \"http://0.0.0.0:8080/\" \"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36\"\n"
     ]
    }
   ],
   "source": [
    "! airflow webserver\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IOStream.flush timed out\n",
      "\u001b[91mUsage:\u001b[0m \u001b[37mairflow\u001b[0m [\u001b[36m-h\u001b[0m] \u001b[36mGROUP_OR_COMMAND\u001b[0m ...\n",
      "\n",
      "\u001b[91mPositional Arguments:\u001b[0m\n",
      "  \u001b[36mGROUP_OR_COMMAND\u001b[0m\n",
      "\n",
      "\u001b[36m    Groups\u001b[0m\n",
      "      \u001b[36mconfig\u001b[0m         \u001b[39mView configuration\u001b[0m\n",
      "      \u001b[36mconnections\u001b[0m    \u001b[39mManage connections\u001b[0m\n",
      "      \u001b[36mdags\u001b[0m           \u001b[39mManage DAGs\u001b[0m\n",
      "      \u001b[36mdb\u001b[0m             \u001b[39mDatabase operations\u001b[0m\n",
      "      \u001b[36mjobs\u001b[0m           \u001b[39mManage jobs\u001b[0m\n",
      "      \u001b[36mpools\u001b[0m          \u001b[39mManage pools\u001b[0m\n",
      "      \u001b[36mproviders\u001b[0m      \u001b[39mDisplay providers\u001b[0m\n",
      "      \u001b[36mroles\u001b[0m          \u001b[39mManage roles\u001b[0m\n",
      "      \u001b[36mtasks\u001b[0m          \u001b[39mManage tasks\u001b[0m\n",
      "      \u001b[36musers\u001b[0m          \u001b[39mManage users\u001b[0m\n",
      "      \u001b[36mvariables\u001b[0m      \u001b[39mManage variables\u001b[0m\n",
      "\n",
      "\u001b[36m    Commands:\u001b[0m\n",
      "      \u001b[36mcheat-sheet\u001b[0m    \u001b[39mDisplay cheat sheet\u001b[0m\n",
      "      \u001b[36mdag-processor\u001b[0m  \u001b[39mStart a standalone Dag Processor instance\u001b[0m\n",
      "      \u001b[36minfo\u001b[0m           \u001b[39mShow information about current Airflow and environment\u001b[0m\n",
      "      \u001b[36mkerberos\u001b[0m       \u001b[39mStart a kerberos ticket renewer\u001b[0m\n",
      "      \u001b[36mplugins\u001b[0m        \u001b[39mDump information about loaded plugins\u001b[0m\n",
      "      \u001b[36mrotate-fernet-key\u001b[0m\n",
      "                     \u001b[39mRotate encrypted connection credentials and variables\u001b[0m\n",
      "      \u001b[36mscheduler\u001b[0m      \u001b[39mStart a scheduler instance\u001b[0m\n",
      "      \u001b[36mstandalone\u001b[0m     \u001b[39mRun an all-in-one copy of Airflow\u001b[0m\n",
      "      \u001b[36msync-perm\u001b[0m      \u001b[39mUpdate permissions for existing roles and optionally DAGs\u001b[0m\n",
      "      \u001b[36mtriggerer\u001b[0m      \u001b[39mStart a triggerer instance\u001b[0m\n",
      "      \u001b[36mversion\u001b[0m        \u001b[39mShow the version\u001b[0m\n",
      "      \u001b[36mwebserver\u001b[0m      \u001b[39mStart a Airflow webserver instance\u001b[0m\n",
      "\n",
      "\u001b[91mOptions:\u001b[0m\n",
      "  \u001b[36m-h\u001b[0m, \u001b[36m--help\u001b[0m         \u001b[39mshow this help message and exit\u001b[0m\n",
      "\n",
      "airflow command error: argument GROUP_OR_COMMAND: invalid choice: 'schedule' (choose from 'cheat-sheet', 'config', 'connections', 'dag-processor', 'dags', 'db', 'info', 'jobs', 'kerberos', 'plugins', 'pools', 'providers', 'roles', 'rotate-fernet-key', 'scheduler', 'standalone', 'sync-perm', 'tasks', 'triggerer', 'users', 'variables', 'version', 'webserver'), see help above.\n"
     ]
    }
   ],
   "source": [
    "! airflow schedule"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
